### ISSUE TITLE
svn gem has invalid ruby syntax

### ISSUE THREAD

**Issue by paleozogt** [codefoundry/svn#1]
```
$ruby --version
ruby 1.8.7 (2012-02-08 patchlevel 358) [i686-darwin10.8.0]
$gem --version
1.8.23
$gem list svn

*** LOCAL GEMS ***

svn (0.1.0)
$irb
1.8.7 :001 > require 'rubygems'
 => true 
1.8.7 :002 > require 'svn'
SyntaxError: /Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:39: syntax error, unexpected tSTAR, expecting tAMPER
        @klass.new( *args, *@added_args )
                            ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:215: syntax error, unexpected kSELF, expecting tAMPER
                *return_ptrs, self, *args, &block
                                  ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:215: Can't change the value of self
                *return_ptrs, self, *args, &block
                                   ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:215: syntax error, unexpected ',', expecting '='
                *return_ptrs, self, *args, &block
                                          ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:219: syntax error, unexpected kSELF, expecting tAMPER
            return_val = meth.call( *return_ptrs, self, *args )
                                                      ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:219: Can't change the value of self
            return_val = meth.call( *return_ptrs, self, *args )
                                                       ^
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:219: syntax error, unexpected ')', expecting '='
/Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn/utils.rb:249: syntax error, unexpected $end, expecting kEND
    from /Users/asimmons/.rvm/rubies/ruby-1.8.7-p358/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:36:in `gem_original_require'
    from /Users/asimmons/.rvm/rubies/ruby-1.8.7-p358/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:36:in `require'
    from /Users/asimmons/.rvm/gems/ruby-1.8.7-p358/gems/svn-0.1.0/lib/svn.rb:4
    from /Users/asimmons/.rvm/rubies/ruby-1.8.7-p358/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:60:in `gem_original_require'
    from /Users/asimmons/.rvm/rubies/ruby-1.8.7-p358/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:60:in `require'
    from (irb):2
1.8.7 :003 > 
```

**Comment by rdblue** [comment]
The problem is that ruby 1.8.7 doesn't allow multiple arguments to be 
splatted in method calls. The solution is something like this to work 
around it:

  return_val = meth.call( *(return_ptrs + args.unshift(self)) )

If you can, I recommend getting around this by using 1.9.2.

rb

**Comment by paleozogt** [comment]
If ruby v1.9.2 is a requirement, please state that in the README or on the github page.

**Comment by rdblue** [comment]
Actually, I intend to fix the problem for 1.8.7 as I described and do a 
point release. But if I eventually decide to drop support for the 1.8 
line, I'll certainly note it in the docs when it happens. My 
recommendation to use 1.9.2 was intended as a work-around to help, not 
upset you.

**Comment by paleozogt** [comment]
Sorry if I came off as brusque-- I wasn't upset.  Just wanted others to know if 1.8.x wasn't supported.  But if you're adding support for 1.8.x, all the better. :)

**Comment by rdblue** [comment]
My mistake, it's hard to interpret tone correctly when dealing with 
only text.

I've made the changes and I'm running the meager tests that I've built 
recently in 1.8.7. I should have a new version out shortly. It'll have 
better error handling, too.

rb

**Comment by rdblue** [comment]
The new gem is released as 0.2.0. I think I cleared up the errors and my tests are all passing in both 1.9.2 and 1.8.7. Thanks for letting me know about the bug.
